test/runtests.jl

```diff
@@ -84,13 +84,21 @@ end
     Xmissingcell = X[inds, :]
     ymissingcell = y[inds]
     @test_throws PosDefException m2 = fit(LinearModel, Xmissingcell, ymissingcell)
-    m2p = fit(LinearModel, Xmissingcell, ymissingcell, true)
+    m2p = fit(LinearModel, Xmissingcell, ymissingcell; allowrankdeficient = true)
     @test isa(m2p.pp.chol, CholeskyPivoted)
     @test rank(m2p.pp.chol) == 11
     @test isapprox(deviance(m2p), 0.2859221258731563)
     @test isapprox(coef(m2p), [0.9178241203127236, 9.089883493902754, 3.01742566831296,
                    4.108734932819495, 4.995249696954908, 6.075962907632594, 0.0, 8.038151489191618,
                    8.848886704358202, 2.8697881579099085, 11.15107375630744, 11.8392578374927])
+
+    m2p_dep = fit(LinearModel, Xmissingcell, ymissingcell, true)
```

Comment:
maybe add a test that this `warn`s?

Reply:
and I guess maybe also the behavior when the kwarg and the positional conflict?

Reply:
Added!